conduit/pipeline/pipeline.go

```diff
@@ -164,6 +164,14 @@ type pipelineImpl struct {
 	pipelineMetadata state
 
 	metricsCallback []conduit.ProvideMetricsFunc
+
+	// preConfigQueue is a pointer to a list of strings that should only be logged
+	// after all configuration of the pipeline is complete (i.e. after init() is successfully called).
+	// The main function of conduit will amass logging strings that will only be output once
+	// everything has been configured.  This is to make it so that the end user sees the direct
+	// error first when launching from the command line. Only after the pipeline has been successfully
+	// configured will log messages be recorded.
+	preConfigQueue *[]string
```

Comment:
I don't know if this is a good idea, but it might be interesting to store `logrus.Entry` here. 

If we generate the log entries with [logger.WithTime](https://github.com/sirupsen/logrus/blob/master/logger.go#L145), when we flush them to the log we'll maintain the time data instead of having everything logged at once when the `init` finishes.